source/common/runtime/runtime_impl.cc

```diff
@@ -26,32 +23,57 @@ namespace Runtime {
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
 std::string RandomGeneratorImpl::uuid() {
-#ifdef __APPLE__
-  // TODO(zuercher): factor out a sensible interface for UUID generation to make it easier
-  // to put a faster implementation in later.
-  char generated_uuid[UUID_LENGTH + 1];
-  uuid_t uuid;
-  uuid_generate_random(uuid);
-  uuid_unparse(uuid, generated_uuid);
-  return std::string(generated_uuid);
-#else
-  int fd = open("/proc/sys/kernel/random/uuid", O_RDONLY);
-  if (-1 == fd) {
-    throw EnvoyException(fmt::format("unable to open uuid, errno: {}", strerror(errno)));
+  // Create UUID from Truly Random or Pseudo-Random Numbers.
+  // See: https://tools.ietf.org/html/rfc4122#section-4.4
+  uint8_t rand[16];
+  RAND_bytes(rand, 16);
```

Comment:
The only reason why `RAND_bytes()` returns anything and isn't declared as `void` is for backward compatibility with OpenSSL. If you look at other functions, i.e. new `void RAND_bytes_with_additional_data()` which `RAND_bytes()` calls internally ([crypto/fipsmodule/rand/rand.c#264](https://boringssl.googlesource.com/boringssl/+/68f84f5c40644e029ed066999448696b01caba7a/crypto/fipsmodule/rand/rand.c#264)), it doesn't return anything, so adding check here will just waste CPU cycles, and it won't be optimized out by the compiler, because we don't do LTO.

I'd agree with you if we linked against system's OpenSSL library, but we're statically linking against known BoringSSL version, so this won't be checking anything that can fail. 

Having said that, I can add `ASSERT()` if you feel strongly about it.